helix-core/src/main/java/org/apache/helix/common/caches/BasicClusterDataCache.java

```diff
@@ -104,6 +108,52 @@ public void refresh(HelixDataAccessor accessor) {
     }
   }
 
+  /**
+   * Selective update Helix Cache by version
+   * @param accessor the HelixDataAccessor
+   * @param reloadKeys keys needs to be reload
+   * @param cachedKeys keys already exists in the cache
+   * @param cachedPropertyMap cached map of propertykey -> property object
+   * @param <T> the type of metadata
+   * @return
+   */
+  public static  <T extends HelixProperty> Map<PropertyKey, T> updateReloadProperties(
```

Comment:
If this is a public static method, why is this in this class?

Reply:
Since this method will be used by different cache class, I think it is better to put this method the basic class of them.